src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,16 +274,28 @@ public void setLayout(Layout layout) {
     }
 
     public void setEntry(BibEntry newEntry) {
-        if(entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.removePropertyChangeListener(this));
-            newEntry.addPropertyChangeListener(this);
+
+        if (entry.isPresent() && (entry.get() != newEntry)) {
+            entry.ifPresent(e -> e.unregisterListener(this));
         }
```

Comment:
could be written simpler: `entry.filter(e -> e != newEntry).ifPresent(e -> e.unregisterListener(this));`